en/lessons/basics/functions.md

```diff
@@ -3,422 +3,234 @@ version: 1.0.0
-To define an anonymous function in Elixir we need the `fn` and `end` keywords.
-Within these we can define any number of parameters and function bodies separated by `->`.
-
-Let's look at a basic example:
+Anonymous functions, or Lambdas, are one of the building blocks of Haskell. They have the following syntax:
 
-```elixir
-iex> sum = fn (a, b) -> a + b end
-iex> sum.(2, 3)
-5
+**Structure of a lambda**
 ```
-
-### The & Shorthand
-
-Using anonymous functions is such a common practice in Elixir there is shorthand for doing so:
-
-```elixir
-iex> sum = &(&1 + &2)
-iex> sum.(2, 3)
-5
+ ╭─ input of the function
+ │ 
+ │ ╭─ Separator between input and output
+ │ │ 
+ │ │  ╭─ output of the function
+\x -> x
 ```
 
-As you probably already guessed, in the shorthand version our parameters are available to us as `&1`, `&2`, `&3`, and so on.
-
-## Pattern Matching
+And incidentally, the example above is one of the most primordial functions: The identity function, or `id`.  
+This function is very simple, as it returns precisely the argument that you pass to it.
 
-Pattern matching isn't limited to just variables in Elixir, it can be applied to function signatures as we will see in this section.
+In Lambdas, the input parameter can be any name, the output can be any expression.
 
-Elixir uses pattern matching to check through all possible match options and select the first matching option to run:
+### Syntactic Sugar: Currying
 
-```elixir
-iex> handle_result = fn
-...>   {:ok, result} -> IO.puts "Handling result..."
-...>   {:ok, _} -> IO.puts "This would be never run as previous will be matched beforehand."
-...>   {:error} -> IO.puts "An error has occurred!"
-...> end
+In Haskell, all your functions have one argument. Inspired by the Lambda Calculus, this design is however extended in
+the language through syntactic sugar that allows you to express yourself better:
 
-iex> some_result = 1
-1
-iex> handle_result.({:ok, some_result})
-Handling result...
-:ok
-iex> handle_result.({:error})
-An error has occurred!
+```haskell
+-- Syntactic sugar
+\x y -> x + y
+ 
+-- Currified version
+\x -> \y -> x + y
 ```
 
-## Named Functions
-
-We can define functions with names so we can easily refer to them later.
-Named functions are defined within a module using the `def` keyword .
-We'll learn more about Modules in the next lessons, for now we'll focus on the named functions alone.
-
-Functions defined within a module are available to other modules for use.
-This is a particularly useful building block in Elixir:
-
-```elixir
-defmodule Greeter do
-  def hello(name) do
-    "Hello, " <> name
-  end
-end
-
-iex> Greeter.hello("Sean")
-"Hello, Sean"
-```
-
-If our function body only spans one line, we can shorten it further with `do:`:
-
-```elixir
-defmodule Greeter do
-  def hello(name), do: "Hello, " <> name
-end
-```
+The term is currification: Your function of two arguments has been transformed behind the scenes into a function
```

Comment:
```suggestion
The term is currification: your function of two arguments has been transformed behind the scenes into a function
```
I believe British English has slightly different capitalisation rules.

Reply:
Yep, only proper nouns and acronyms are capitalised after a colon in en_UK